src/types/Client.d.ts

```diff
@@ -13,7 +15,7 @@ export interface ClientConfig {
   port?: number
   timeout?: number
   queryTimeout?: number
-  observer?: (res: RequestResult, client: Client) => void
+  observer?: <T extends object = object>(res: RequestResult<T | errors.FaunaHTTPError>, client: Client) => void
```

Comment:
Same for `object` type restriction as mentioned below

Reply:
not really. observer accepts the full response or error